webapp/src/tutorialCodeValidation.tsx

```diff
@@ -38,15 +39,25 @@ export class MoveOn extends data.Component<TutorialCodeValidationProps, tutorial
         this.props.onNoButtonClick();
     }
 
+    compileMessage(rules: pxt.tutorial.TutorialRuleStatus[]): string {
+        let s: string = "";
+        if (rules != undefined) {
+            for (let i = 0; i < rules.length; i++) {
+                if (rules[i].RuleTurnOn && !rules[i].RuleStatus) {
+                    s = s.concat(s, rules[i].RuleMessage);
+                }
+            }
+        }
+        return s;
+    }
+
     renderCore() {
         const vis = this.props.isTutorialCodeInvalid;
+        const message = this.compileMessage(this.props.ruleComponents);
```

Comment:
you can iterate over the messages inside the react component using `.map` (see https://github.com/microsoft/pxt/blob/master/webapp/src/errorList.tsx#L150); concatenating the strings means there won't be line breaks between messages, and you won't be able to adjust the styles/spacing